pkg/deployment/rotation/arangod.go

```diff
@@ -52,8 +55,24 @@ func affinityCompare(_ api.DeploymentSpec, _ api.ServerGroup, spec, status *core
 				e = err
 				return
 			} else if specC != statusC {
-				status.Affinity = spec.Affinity.DeepCopy()
 				mode = mode.And(SilentRotation)
+
+				var specArchs api.ArangoDeploymentArchitecture
+				var statusArchs api.ArangoDeploymentArchitecture
+
+				if spec.Affinity != nil && spec.Affinity.NodeAffinity != nil && spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+					specArchs = api.GetArchsFromNodeSelector(spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms)
+				}
+
+				if status.Affinity != nil && status.Affinity.NodeAffinity != nil && status.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution != nil {
+					statusArchs = api.GetArchsFromNodeSelector(status.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms)
+				}
+
+				if !reflect.DeepEqual(specArchs, statusArchs) {
```

Comment:
Maybe it is possible to move all this new code to a separate function/method? 
e.g.
```
mode = addGracefulRotationIfNeeded(mode, spec, api)
```